infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -226,12 +233,15 @@ protected void internalStart(Map<String, String> startOptions) throws Infrastruc
                 // see comments above why executor is explicitly put into arguments
                 .thenComposeAsync(checkFailure(startFailure), executor)
                 .thenCompose(setContext(currentContext, checkServers(toCancelFutures, machine)))
-                .thenRun(finishStartupTracingSpan(machineName))
                 .exceptionally(publishFailedStatusAndRecordFailureTrace(startFailure, machineName));
         machinesFutures.put(machineName, machineBootChain);
       }
 
       waitMachines(machinesFutures, toCancelFutures, startFailure);
+
+      if (waitRunningAsyncSpan != null) {
+        waitRunningAsyncSpan.close();
```

Comment:
Should we close it in any case or only in a successful case?

Reply:
Moved to `TracerUtil`

Reply:
Good catch, reworked this. It's a minor hassle since closing a scope twice is undefined behavior, so the close call is now in the finally and we can add a failure status if necessary.